File: react-news/common/component-resolver.js
import React, { Component } from 'react';

import ArticleText from '../components/article/article-text';
import ArticleImage from '../components/article/article-image';
import ByLine from '../components/article/article-byline';

export default class ComponentResolver {
    renderComponent(componentData, key) {
        switch (componentData.type) {
            case 'text' : return (
                <ArticleText data={componentData} key={key} />  
            )
            case 'image' : return (<ArticleImage data={componentData} key={key} />)
            case 'byline' : return (<ByLine data={componentData} key={key} />)
        }
        return null;
    }
}

File: xamarin-news/xam-android-news/ArticleLogic/ComponentResolver.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.Content;
using xam_android_news.Views.Article;
using Android.Views;

namespace xam_android_news.ArticleLogic
{
    /// <summary>
    /// Maps JSON data to Component Views
    /// </summary>
    public class ComponentResolver
    {
        LayoutInflater Inflater;
        public ComponentResolver(Context context)
        {
            Inflater = (LayoutInflater) context.GetSystemService(Context.LayoutInflaterService);
        }

        public ArticleComponent GetComponentFor(Context context, dynamic componentSegment)
        {
            string type = componentSegment.type;
            switch (type)
            {
                case "text": return ArticleTextView.Create(Inflater, componentSegment);
                case "image": return ImageComponent.Create(Inflater);
                case "byline": return BylineComponent.Create(Inflater);
                default: return new BlankComponent(context);
            }
        }
    }
}